File: app/src/main/java/com/olodjinha/guilherme/alodjinha/fragment/BannerFragment.java
package com.olodjinha.guilherme.alodjinha.fragment;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.bumptech.glide.Glide;
import com.olodjinha.guilherme.alodjinha.R;
import com.olodjinha.guilherme.alodjinha.model.Banner;

/**
 * Created by Guilherme on 19/06/2018.
 */

public class BannerFragment extends Fragment {

    private LinearLayout linearLayoutBanners;
    private ImageView imageViewBanner;
    private Banner detail;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_banner, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        linearLayoutBanners = view.findViewById(R.id.linearLayoutBanners);
        imageViewBanner = view.findViewById(R.id.imageViewBanner);

        Glide.with(getContext()).load(getDetail().getUrlImagem()).into(imageViewBanner);

        linearLayoutBanners.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.submarino.com.br"));
                startActivity(browserIntent);
            }
        });
    }

    public Banner getDetail() {
        return detail;
    }

    public void setDetail(Banner detail) {
        this.detail = detail;
    }
}

File: o-lodjinha/app/src/main/java/com/sumiya/olodjinha/UI/Fragments/BannerFragment.kt
package com.sumiya.olodjinha.ui.fragments

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.sumiya.olodjinha.R
import com.sumiya.olodjinha.model.BannerModel
import kotlinx.android.synthetic.main.fragment_banner.*

class BannerFragment : Fragment() {

    private var listener: BannerListener? = null
    lateinit var banner: BannerModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_banner, container, false)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is BannerListener) {
            listener = context
        } else {
            throw RuntimeException(context.toString() + " must implement BannerListener")
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        Glide
                .with(this)
                .load(banner.urlImagem)
                .into(bannerImage)

        bannerImage.setOnClickListener {
            listener?.bannerCLick(banner.linkUrl)
        }
    }

    override fun onDetach() {
        super.onDetach()
        listener = null
    }

    interface BannerListener {
        fun bannerCLick(bannerUrl: String)
    }
}
